app/graphql/types/rule.rb

```diff
@@ -14,24 +14,57 @@ class Rule < Types::BaseObject
     field :severity, String, null: false
     field :remediation_available, Boolean, null: false
     field :profiles, [::Types::Profile], null: true
-    field :identifier, ::Types::RuleIdentifier, null: true
-    field :references, [::Types::RuleReference], null: true,
-                                                 extras: [:lookahead]
+    field :identifier, String, null: true
+    field :references, [String], null: true
     field :compliant, Boolean, null: false do
-      argument :system_id, String, 'Is a system compliant?', required: true
+      argument :system_id, String, 'Is a system compliant?', required: false
+      argument :profile_id, String, 'Is a system compliant with this profile?', required: false
     end
 
-    def compliant(system_id:)
-      object.compliant?(Host.find(system_id))
+    def compliant(args = {})
+      return false unless system_id(args) && profile_id(args) && context[:rule_results][object.id][profile_id(args)].present?
+
+      latest_result = context[:rule_results][object.id][profile_id(args)]
+      %w[pass notapplicable notselected].include? latest_result
+    end
+
+    def references
+      if context[:"rule_references_#{object.id}"].nil?
+        ::CollectionLoader.for(::Rule, :rule_references).load(object).then do |references|
+          references
+        end
+      else
+        ::RecordLoader.for(::RuleReference)
+          .load_many(context[:"rule_references_#{object.id}"])
+          .then do |references|
+          references.map { |ref| [ref.href, ref.label] }
```

Comment:
It's insane how much faster this is - in a request that took 10s, it takes 5s now, all those 5 seconds were spent serializing `::Type::RuleReference` instances by GraphQL-ruby